api/python/quilt3/packages.py

```diff
@@ -1360,26 +1377,59 @@ def push(
         workflow=..., force: bool = False, dedupe: bool = False
     ):
         """
-        Copies objects to path, then creates a new package that points to those objects.
-        Copies each object in this package to path according to logical key structure,
-        then adds to the registry a serialized version of this package with
-        physical keys that point to the new copies.
+        Creates a new package, or a new revision of an existing package in a
+        package registry in Amazon S3.
 
-        Note that push is careful to not push data unnecessarily. To illustrate, imagine you have
-        a PackageEntry: `pkg["entry_1"].physical_key = "/tmp/package_entry_1.json"`
+        By default, local files are copied to the destination S3 bucket and path
+        according to logical key structure. After any local objects are copied,
+        a new package manifest is created that points to the objects in their
+        new locations.
 
-        If that entry would be pushed to `s3://bucket/prefix/entry_1.json`, but
-        `s3://bucket/prefix/entry_1.json` already contains the exact same bytes as
-        '/tmp/package_entry_1.json', `quilt3` will not push the bytes to s3, no matter what
-        `selector_fn('entry_1', pkg["entry_1"])` returns.
+        The optional parameter `selector_fn` allows callers to choose which
+        files are copied to the destination bucket, and which retain their
+        existing physical key. When using selector functions, it is important to
+        always copy local files to S3, otherwise the resulting package will be
+        inaccessible to users accessing it from Amazon S3.
+
+        The Package class includes two additional built-in selector functions:
+
+        * `Package.selector_fn_copy_all` copies all files to the destination path.
+        * `Package.selector_fn_copy_local` copies only local files to the
+          destination path. Any PackageEntry's with physical keys pointing to
+          objects in other buckets will retain their existing physical keys in
+          the resulting package.
+
+        If we have a package with entries:
+
+        * `pkg["entry_1"].physical_key = s3://bucket1/user/pkg_name/entry_1`
```

Comment:
might make sense to use different key to avoid giving impression we preserve key here